src/validation/__tests__/ProhibitDeprecatedFieldsRule-test.js

```diff
@@ -0,0 +1,135 @@
+// @flow strict
+
+import { describe, it } from 'mocha';
+
+import { buildSchema } from '../../utilities/buildASTSchema';
+
+import { ProhibitDeprecatedFieldsRule } from '../rules/optional/ProhibitDeprecatedFieldsRule';
+
+import { expectValidationErrorsWithSchema } from './harness';
+
+const schema = buildSchema(`
+  enum EnumType {
+    NORMAL_VALUE
+    DEPRECATED_VALUE @deprecated(reason: "Some enum reason.")
+    DEPRECATED_VALUE_WITH_NO_REASON @deprecated
+  }
+
+  type Query {
+    normalField(enumArg: [EnumType]): String
+    deprecatedField: String @deprecated(reason: "Some field reason.")
+    deprecatedFieldWithNoReason: String @deprecated
+  }
+`);
+
+function expectErrors(queryStr) {
+  return expectValidationErrorsWithSchema(
+    schema,
+    ProhibitDeprecatedFieldsRule,
+    queryStr,
+  );
+}
+
+function expectValid(queryStr) {
+  expectErrors(queryStr).to.deep.equal([]);
+}
```

Comment:
nit: these two functions kind of boring since you see them in every validation tests so it's better to put `schema` closer to actual tests so you do less scrolling.
I always put boring and repairing parts closer to imports and more specific stuff closer to tests